cpp/src/groupby/hash/compute_mapping_indices.cuh

```diff
@@ -125,6 +125,7 @@ CUDF_KERNEL void mapping_indices_kernel(size_type num_input_rows,
     block.sync();
```

Comment:
Is this sync still necessary?

Reply:
if it's not executed by all threads in a block, calling a block sync here is problematic. 

you may want to do something like 
```cpp
// all threads in the block will fall into the same code path
for (auto idx = cudf::detail::grid_1d::global_thread_id(); idx - block.thread() < num_input_rows; idx += stride)  {
  if (idx < num_input_rows) { find_local_mapping(...); }
  block.sync(); // this block sync is safe
  ...
}
```
then the sync you added in this PR should be no longer necessary.

Reply:
Done.

Reply:
It's not necessary. That function already handles such out of bound cases.